quickwit-core/src/indexing/split.rs

```diff
@@ -127,17 +131,43 @@ impl Split {
             num_parsing_errors: 0,
             storage,
             metastore,
+            timestamp_field,
         })
     }
 
     /// Add document to the index split.
     pub fn add_document(&mut self, doc: Document) -> anyhow::Result<()> {
-        //TODO: handle time range when docMapper is available
+        let mut computed_time_range = None;
+        if let Some(timestamp_field) = self.timestamp_field {
+            let mut split_time_range = self
+                .metadata
+                .time_range
+                .as_ref()
+                .and_then(|range| Some(range.start..range.end))
+                .unwrap_or((i64::MAX)..(i64::MIN));
+
+            //TODO: discuss finding a better way to extract the timestamp_field value from
+            // the document
+            if let Some(timestamp) = doc
+                .get_first(timestamp_field)
+                .and_then(|field_value| field_value.i64_value())
+            {
+                if timestamp < split_time_range.start {
+                    split_time_range.start = timestamp;
+                }
+                if timestamp > split_time_range.end {
+                    split_time_range.end = timestamp;
+                }
+            }
+            computed_time_range = Some(split_time_range);
+        }
+
         self.index_writer
             .as_ref()
             .ok_or_else(|| anyhow::anyhow!("Missing index writer."))?
             .add_document(doc);
         self.metadata.num_records += 1;
```

Comment:
I would only do the assignment in the if condition. 
Here we are saying that if one document has no timestamp information, we delete the information for the split

Reply:
This is fine now and as discussed, I now have a function to perform the time_range computation.